nifi-commons/nifi-property-utils/src/main/java/org/apache/nifi/properties/ProtectedPropertyContext.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.properties;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+/**
+ * A context for protected properties, encapsulating the name and location of a property.
+ * @see PropertyLocation#contextFor(String)
+ */
+public class ProtectedPropertyContext {
+    /**
+     * A property location, representing a specific configuration file.
+     */
+    public enum PropertyLocation {
+        LOGIN_IDENTITY_PROVIDERS("login-identity-providers.xml"),
+        AUTHORIZERS("authorizers.xml"),
+        NIFI_PROPERTIES("nifi.properties"),
+        NIFI_REGISTRY_PROPERTIES("nifi-registry.properties"),
+        CUSTOM("custom");
```

Comment:
Although this looks like it should cover existing configuration files, it seems somewhat difficult to maintain. Without making this too abstract, what do you think about other options for configuring the context?  Perhaps separating the sensitive properties provider configuration into its own file where these mappings could be configured?